### PR TITLE
Fix 252

### PR DESCRIPTION
**Pull request by almasaeed2010**
@mboudet could you please take a look at this solution to verify it works. I don't have a tripal 2 site right now so it may be faster for you to test. 

Thanks!

### CODE REVIEW

File: includes/feature_heatmap_form.inc [hunk 1]
```diff
@@ -181,12 +181,14 @@ function generate_heatmap_values() {
...
     $text = array();
     $max_id_len = 0;
+    global $base_url;
     foreach ($expression_data as $feature_id => $values) {
       // obtain max feature id length
       if (strlen($feature_id) > $max_id_len) {
         $max_id_len = strlen($feature_id);
       }
-      $y[] = '<a href="/feature/' . $feature_id . '">' . $feature_id . '</a>';
+      $nid = chado_get_nid_from_id('feature', $feature_id);
```
Comment (mboudet) [inline includes/feature_heatmap_form.inc]: The issue is that 'feature_id' is NOT actually the feature ID. 
$feature_id is set with 
`$feature_ids = preg_split("/[\s,]+/", $heatmap_feature_uniquename, -1, PREG_SPLIT_NO_EMPTY);`
So it's basically the unique name of the feature. So 'chado_get_nid_from_id' fails.

Comment (almasaeed2010) [reply includes/feature_heatmap_form.inc]: hmmm good catch. I think the tripal 3 version does a much better job at linking out. I'll take a look at it there and see if we can port back.

**Comment by almasaeed2010** [comment]
@mboudet the latest commit should fix the issue using the correct feature id.

**Comment by mboudet** [comment]
It should work, but the link points to "/node/" , as if the nid was not found. I'm looking into it.

File: includes/feature_heatmap_form.inc [hunk 2]
```diff
@@ -186,7 +186,9 @@ function generate_heatmap_values() {
       if (strlen($feature_id) > $max_id_len) {
         $max_id_len = strlen($feature_id);
       }
-      $y[] = '<a href="/feature/' . $feature_id . '">' . $feature_id . '</a>';
+      $fid = $expressions[$feature_id]->feature_id;
```
Comment (mboudet) [inline includes/feature_heatmap_form.inc]: Ok, got it. $expressions[$feature_id] is an array (one value for each biomaterial), so we need to take the first element of the array, and then get the ->feature_id. (All elements will have the same feature_id, so might as well take the first)

We can probably just call reset($expressions[$feature_id])->feature_id to get the value.

Comment (almasaeed2010) [reply includes/feature_heatmap_form.inc]: Oh that's a good catch! I updated the code to use reset. Thanks!

**Comment by mboudet** [comment]
Works fine for me.

**Comment by almasaeed2010** [comment]
Thanks!

**Pull request merged**